formats/json/jvmMain/src/kotlinx/serialization/json/JvmStreams.kt

```diff
@@ -80,3 +80,93 @@ public fun <T> Json.decodeFromStream(
 @ExperimentalSerializationApi
 public inline fun <reified T> Json.decodeFromStream(stream: InputStream): T =
     decodeFromStream(serializersModule.serializer(), stream)
+
+/**
+ * Description of [decodeToSequence]'s JSON input shape.
+ *
+ * The sequence represents a stream of objects parsed one by one;
+ * [LazyStreamingFormat] defines a separator between these objects.
+ * Normally, these objects are not separated by meaningful characters ([WHITESPACE_SEPARATED])
+ * or the whole stream is a large array of objects separated with commas ([ARRAY_WRAPPED]).
+ */
+@ExperimentalSerializationApi
+public enum class LazyStreamingFormat {
```

Comment:
I'm wondering whether the name should include Json as an indicator that this is specific to Json. Alternatively a generic class (not in the json package) may make sense to be used by multiple formats (but should then be defined a bit more generically). The issue I'm considering is multiple formats in the same code base where both formats support stream decoding (eg. ktor content negotiation).

Reply:
The name is not final for now, so we can come up with a better one. Although I don't think that making it generic would be a good idea because whitespace vs array wrapping looks like the choice only for json.

Reply:
@pdvrieze it could be done, but alas, it still will be JSON-centric as our main target here.
Making it more generalized will only confuse `Json` format users, but will bring insignificant benefits for other formats.
Unlike, for example, `SerialDescriptor`, there shouldn't be much value in sharing this mode between e.g. XML and JSON formats

Reply:
@qwwdfsad My main intuition is that the format would probably be format specific. But the concept would work for XML (either separate xml messages - I know this is technically nonconformant - that never stops people from wanting to parse/generate almostXML, or a wrapper "list"). The current name (in a different package) would also make sense in the XML case (and other formats), which means that systems that support both have naming conflicts (renaming imports would fix this).

As a different aside, perhaps **`...Format`** has challenges given what normally format means in the context of the library.